### PR TITLE
refactor: replace zap with slog to log traceID

### PR DESCRIPTION
**Pull request by theSuess**

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -13,29 +13,28 @@ import (
...
 
 	// Create an HTTP client configured from env vars.
 	// If no specific env vars are set, this will return a http client that does not perform any retries.
 	httpCli := clientFromEnv()
 
-	server, err := qphttp.NewServer(globalLogger)
+	server, err := qphttp.NewServer()
 	if err != nil {
-		globalLogger.Fatal("Cannot create server", zap.Error(err))
+		slog.Error("Cannot create server", "err", err)
```
Comment (roobre) [inline cmd/main.go:37]: Seems like `os.Exit()` is missing here

File: cmd/main.go [hunk 2]
```diff
@@ -108,10 +108,10 @@ func main() {
 	}
 
 	listen := ":3333"
-	globalLogger.Info("Starting QuickPizza", zap.String("listenAddress", listen))
+	slog.Info("Starting QuickPizza", "listenAddress", listen)
 	err = http.ListenAndServe(listen, server)
 	if err != nil {
-		globalLogger.Error("Running HTTP server", zap.Error(err))
+		slog.Error("Running HTTP server", "err", err)
```
Comment (roobre) [inline cmd/main.go:114]: We probably want `os.Exit` here as well. The original line was not `Fatal` but I think it should always have been 🌍🧑‍🚀🔫 :)

**Review by roobre — COMMENTED** [review]
Looking good!

Other than the two small nits I've mentioned, when running some tests I noticed that `go.mod` et. al do not seem to be clean. Could you run `go mod tidy && go mod vendor` and update the PR?

Thank you!

**Comment by theSuess** [comment]
Should all be resolved now!

**Review by roobre — APPROVED** [review]

**Pull request merged**